validation/src/main/java/io/serverlessworkflow/validation/SwitchStateValidator.java

```diff
@@ -0,0 +1,163 @@
+package io.serverlessworkflow.validation;
+
+import io.serverlessworkflow.api.Workflow;
+import io.serverlessworkflow.api.defaultdef.DefaultConditionDefinition;
+import io.serverlessworkflow.api.interfaces.State;
+import io.serverlessworkflow.api.states.SwitchState;
+import io.serverlessworkflow.api.timeouts.TimeoutsDefinition;
+import io.serverlessworkflow.api.validation.ValidationError;
+
+import java.time.Duration;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+
+class SwitchStateValidator extends CommonStateValidator<SwitchState> {
+
+    private static final String NON_NEGATIVE_DURATION_MUST_BE_PROVIDED =
+            "When configured, it must be set with a greater than zero ISO 8601 time duration. For example PT30S.";
+
+    static final String INVALID_EVENT_TIMEOUT_ERROR =
+            "An invalid \"eventTimeout\": \"%s\" configuration was provided for the serverless workflow: \"%s\". "
+                    + NON_NEGATIVE_DURATION_MUST_BE_PROVIDED;
+
+    static final String EVENT_TIMEOUT_REQUIRED_ERROR =
+            "The \"eventTimeout\" configuration is required for the \"eventConditions\" based switch state \"%s\" that belongs to the serverless workflow: \"%s\".";
+
+    static final String DATA_CONDITIONS_AND_EVENT_CONDITIONS_FOUND_ERROR =
+            "DataConditions and eventConditions where found at the same time for the switch state \"%s\".";
+
+    static final String NEXT_STATE_NOT_FOUND_FOR_DEFAULT_CONDITION_ERROR =
+            "The \"nextState\" : \"%s\" configured for the \"defaultCondition\" transition in the switch state \"%s\", was not found in the serverless workflow: \"%s\".";
+
+    static final String NEXT_STATE_REQUIRED_FOR_DEFAULT_CONDITION_ERROR =
+            "The \"nextState\" is required for the \"defaultCondition\" transition in the switch state \"%s\" that belongs to the serverless workflow: \"%s\".";
+
+    @Override
+    protected List<ValidationError> runSpecificValidations(Workflow workflow, SwitchState state) {
+        List<ValidationError> errors = new ArrayList<>();
+
+        validateDefaultCondition(workflow, state).ifPresent(errors::add);
+        validateEventCondition(workflow, state).ifPresent(errors::add);
+        errors.addAll(validateDataConditionAndEventConditionPresent(state));
+
+        return errors;
+    }
+
+    private static List<ValidationError> validateDataConditionAndEventConditionPresent(
+            SwitchState switchState) {
+        List<ValidationError> errors = new ArrayList<>();
+
+        if (!switchState.getDataConditions().isEmpty() && !switchState.getEventConditions().isEmpty()) {
+            ValidationError error = new ValidationError();
```

Comment:
You already have a protected final method in the parent class to create the validation error